src/app/dao/tests/ChatDAO.spec.ts

```diff
@@ -1,17 +1,259 @@
+        await joinerDAO.set(id, { ...JoinerMocks.INITIAL, creator });
+        return id;
+    }
     beforeEach(async() => {
         await setupEmulators();
-        dao = TestBed.inject(ChatDAO);
+        chatDAO = TestBed.inject(ChatDAO);
+        partDAO = TestBed.inject(PartDAO);
+        joinerDAO = TestBed.inject(JoinerDAO);
     });
     it('should be created', () => {
-        expect(dao).toBeTruthy();
+        expect(chatDAO).toBeTruthy();
+    });
+    describe('subscribeToMessages', () => {
+        it('should rely on observingWhere of messages DAO and sort by postedTime', () => {
+            // Given a chat DAO
+            const messagesDAO: IFirebaseFirestoreDAO<Message> = chatDAO.subCollectionDAO('chatId', 'messages');
+            spyOn(messagesDAO, 'observingWhere');
+            // When calling subscribeToMessages
+            const callback: FirebaseCollectionObserver<Message> =
+                new FirebaseCollectionObserver<Message>(() => {}, () => {}, () => {});
+            chatDAO.subscribeToMessages('chatId', callback);
+            // Then it should call observingWhere and sort by postedTime
+            expect(messagesDAO.observingWhere).toHaveBeenCalledOnceWith([], callback, 'postedTime');
+        });
+    });
+    describe('on any chat', () => {
+        let myUser: MinimalUser;
+        let myMessageId: string;
+        let otherUser: MinimalUser;
+        let otherMessageId: string;
+        beforeEach(async() => {
+            // Given a user
+            const other: FireAuth.User = await createConnectedGoogleUser(true, 'foo@bar.com', 'other-user');
+            // and a chat (here the lobby, but this could be any chat)
+            await chatDAO.set('lobby', {});
+            // with a message from another user
+            otherUser = {
+                name: 'other-user',
+                id: other.uid,
+            };
+            const message: Message = {
+                content: 'hullo there',
+                sender: otherUser,
+                postedTime: serverTimestamp(),
+            };
+            otherMessageId = await chatDAO.addMessage('lobby', message);
+            await signOut();
+            // and a message from the current user, who is able to add messages
+            const user: FireAuth.User = await createConnectedGoogleUser(true, 'foo@bar.com', 'user');
+            myUser = {
+                name: 'user',
+                id: user.uid,
+            };
+            myMessageId = await chatDAO.addMessage('lobby', { ...message, sender: myUser });
+        });
+        it('should forbid disconnected users to read a chat', async() => {
+            // Given a disconnected user
+            await signOut();
+            // When trying to read a chat
+            const chatRead: Promise<MGPOptional<Chat>> = chatDAO.read('lobby');
+            // Then it fails
+            await expectAsync(chatRead).toBeRejected();
```

Comment:
Y a un moyen de faire "toBeRejectWithError(string)" ? J'avais essayé ça j'ai dû faire d'horribles sparadraps.
Et les expect to fail flous, ça me semble moyen :D

Reply:
un peu plus saignant du cul mais fait ! (ici on veut pas checker le message car il dépend du numéro de ligne où se trouve la règle du firestore.json qui limite l'accès, donc ça voudrait dire changer TOUS ces tests à chaque changement du firestore.json D:) Donc on check maintenant qu'on a bien une "permission-denied" de Firestore